lib/adapters/gun.ex

```diff
@@ -101,23 +100,27 @@ if Code.ensure_loaded?(:gun) do
     end
 
     def handle_info(
-          {:DOWN, _ref, :process, conn, reason},
-          %State{connection: conn, receiver: receiver} = state
+          {:gun_down, _, _, reason, _, _},
```

Comment:
There's one more message I think we should handle: https://ninenines.eu/docs/en/gun/1.3/manual/gun_error/ Not sure though what kind of errors it returns, but I guess it can be handled in the same way